src/git.md

```diff
@@ -128,6 +128,33 @@ git remote set-url origin <URL>
 
 where the `<URL>` is your new fork.
 
+### I changed a submodule by accident
+
+Usually people notice this when rustbot posts a comment on github that `cargo` has been modified:
+
+![rustbot submodule comment](./img/rustbot-submodules.png)
+
+You might also notice conflicts in the web UI:
+
+![conflict in src/tools/cargo](./img/submodule-conflicts.png)
```

Comment:
@jyn514 
I was annoyed by this for some time, now I use this `justfile` for [just](https://github.com/casey/just) to rebase or sync master:

```
old_branch := `git rev-parse --abbrev-ref HEAD`
rebm:
        git stash
        git checkout up-master
        git pull
        git submodule sync
        git submodule update
        git stash
        git checkout {{old_branch}}
        git rebase up-master

sync:
        git stash
        git checkout up-master
        git pull
        git submodule sync
        git submodule update
```

Then only run `j rebm` to to rebase master.
I'm not sure whether we can provide some handy functions to help developers, but I'm sure for those devs don't touch submodules, it will save a lot of time to fix submodule conflicts.

Reply:
Leaving comments on closed PRs is not a good way to contact me, I have too many github notifications to read them all.

It sounds like you have a suggestion to improve this section :) feel free to open either an issue or a PR.